adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -1292,6 +1327,34 @@ trait MultisampleGenomicRDD[T, U <: MultisampleGenomicRDD[T, U]] extends Genomic
    * The samples who have data contained in this GenomicRDD.
    */
   val samples: Seq[Sample]
+
+  /**
+   * Replaces the sample metadata attached to the RDD.
+   *
+   * @param newSamples The new sample metadata to attach.
+   * @return A GenomicRDD with new sample metadata.
+   */
+  def replaceSamples(newSamples: Seq[Sample]): U
```

Comment:
Ah, good point. My preference is that we keep the type of the field `samples` as a `Seq` (because `Iterable` doesn't support indexed lookup, and it is useful to be able to easily query "which sample is sample 100?") but for these functions, `Iterable` is preferable.